modules/core/src/v2/coins/avaxc.ts

```diff
@@ -1,21 +1,236 @@
+import { getBuilder, Eth, AvaxC as AvaxCAccountLib } from '@bitgo/account-lib';
+import * as common from '../../common';
+
+import { optionalDeps } from './eth';
+import { Erc20Token } from './erc20Token';
+import { Wallet } from '../wallet';
+
+// For precreateBitgo
+interface PrecreateBitGoOptions {
+  enterprise?: string;
+  newFeeAddress?: string;
+}
+
+// For explainTransaction
+export interface ExplainTransactionOptions {
+  txHex?: string;
+  halfSigned?: {
+    txHex: string;
+  };
+  feeInfo: TransactionFee;
+}
+
+interface AvaxcTransactionParams extends TransactionParams {
+  gasPrice?: number;
+  gasLimit?: number;
+  hopParams?: HopParams;
+  hop?: boolean;
+}
+
+interface VerifyAvaxcTransactionOptions extends VerifyTransactionOptions {
+  txPrebuild: TransactionPrebuild;
+  txParams: AvaxcTransactionParams;
+}
+
+// For preSign
+interface PresignTransactionOptions extends TransactionPrebuild, BasePresignTransactionOptions {
+  wallet: Wallet;
+}
+
+export interface TransactionPrebuild extends BaseTransactionPrebuild {
+  hopTransaction?: HopPrebuild;
+  buildParams: {
+    recipients: Recipient[];
+  };
+  recipients: TransactionRecipient[];
+  nextContractSequenceId: string;
+  gasPrice: number;
+  gasLimit: number;
+  isBatch: boolean;
+  coin: string;
+  token?: string;
+}
+
+// For createHopTransactionParams
+interface HopTransactionBuildOptions {
+  wallet: Wallet;
+  recipients: Recipient[];
+  walletPassphrase: string;
+}
+
+interface Recipient {
+  address: string;
+  amount: string;
+  data?: string;
+}
+
+// For getExtraPrebuildParams
+interface BuildOptions {
+  hop?: boolean;
+  wallet?: Wallet;
+  recipients?: Recipient[];
+  walletPassphrase?: string;
+  [index: string]: unknown;
+}
+
+// For FeeEstimate
+interface FeeEstimate {
+  gasLimitEstimate: number;
+  feeEstimate: number;
+}
+
+/**
+ * The extra parameters to send to platform build route for hop transactions
+ */
+interface HopParams {
+  hopParams: {
+    gasPriceMax: number;
+    userReqSig: string;
+    paymentId: string;
+  };
+  gasLimit: number;
+}
+
+/**
+ * The prebuilt hop transaction returned from the HSM
+ */
+interface HopPrebuild {
+  tx: string;
+  id: string;
+  signature: string;
+  paymentId: string;
+  gasPrice: number;
+  gasLimit: number;
+  amount: number;
+  recipient: string;
+  nonce: number;
+  userReqSig: string;
+  gasPriceMax: number;
+}
+
+// For txPreBuild
+export interface TxInfo {
+  recipients: Recipient[];
+  from: string;
+  txid: string;
+}
+
+export interface EthTransactionFee {
+  fee: string;
+  gasLimit?: string;
+}
+
+export interface TxPreBuild extends BaseTransactionPrebuild {
+  txHex: string;
+  txInfo: TxInfo;
+  feeInfo: EthTransactionFee;
+  source: string;
+  dataToSign: string;
+  nextContractSequenceId?: string;
+  expireTime?: number;
+  hopTransaction?: string;
+}
+
+// For signTransaction
+export interface SignTransactionOptions extends BaseSignTransactionOptions {
```

Comment:
rename this type to AvaxSignTransactionOptions like https://github.com/BitGo/BitGoJS/blob/e0c18a15a72d0ccc1b1997f779d0a18ae0559faa/modules/core/src/v2/coins/abstractEthLikeCoin.ts#L28-L28

Reply:
sure